contracts/exchangeIssuance/FlashMintHyETH.sol

```diff
@@ -0,0 +1,641 @@
+            }
+            return;
+        }
+        if (_isInstadapp(_component)) {
+            _depositIntoInstadapp(IERC4626(_component), _amount);
+            return;
+        }
+        IPendleStandardizedYield syToken = _getSyToken(IPendlePrincipalToken(_component));
+        if (syToken != IPendleStandardizedYield(address(0))) {
+            _depositIntoPendle(IPendlePrincipalToken(_component), _amount, syToken);
+            return;
+        }
+        if (IERC20(_component) == acrossToken) {
+            _depositIntoAcross(_amount);
+            return;
+        }
+        if (_component == dexAdapter.weth) {
+            IWETH(dexAdapter.weth).deposit{ value: _amount }();
+            return;
+        }
+        revert("Missing Swapdata for non-standard component");
+    }
+
+    /**
+     * @dev Withdraw ETH from given component
+     *
+     */
+    function _withdrawFromComponent(
+        address _component,
+        uint256 _amount,
+        DEXAdapterV2.SwapData memory _swapData
+    ) internal {
+        if(_swapData.exchange != DEXAdapterV2.Exchange.None) {
+            require(_swapData.path.length > 1, "zero length swap path");
+            require(_swapData.path[0] == _component, "Invalid input token");
+            require(_swapData.path[_swapData.path.length - 1] == DEXAdapterV2.ETH_ADDRESS || _swapData.path[_swapData.path.length - 1] == dexAdapter.weth, "Invalid output token");
+            uint256 ethReceived = dexAdapter.swapExactTokensForTokens(_amount, 0, _swapData);
+            if(_swapData.path[_swapData.path.length - 1] == dexAdapter.weth) {
+                IWETH(dexAdapter.weth).withdraw(ethReceived);
+            }
+            return;
+        }
+        if (_isInstadapp(_component)) {
+            _withdrawFromInstadapp(IERC4626(_component), _amount);
+            return;
+        }
+        IPendleMarketV3 market = pendleMarkets[IPendlePrincipalToken(_component)];
+        if (market != IPendleMarketV3(address(0))) {
+            _withdrawFromPendle(IPendlePrincipalToken(_component), _amount, market);
+            return;
+        }
+        if (IERC20(_component) == acrossToken) {
+            _withdrawFromAcross(_amount);
+            return;
+        }
+        if (_component == dexAdapter.weth) {
+            IWETH(dexAdapter.weth).withdraw(_amount);
+            return;
+        }
+        revert("Missing Swapdata for non-standard component");
+    }
+
+    /**
+     * @dev Deposit eth into steth and then into instadapp vault
+     *
+     */
+    function _depositIntoInstadapp(IERC4626 _vault, uint256 _amount) internal {
+        uint256 stETHAmount = _vault.previewMint(_amount);
+        _depositIntoLido(stETHAmount);
+        _vault.mint(_amount, address(this));
+    }
+
+    /**
+     * @dev Deposit eth into steth
+     *
+     */
+    function _depositIntoLido(uint256 _amount) internal {
+        stETH.submit{ value: _amount }(address(0)); // TODO: Check if we want to pass referral address
```

Comment:
nit: remove TODO

no need for referral address

Reply:
removed here [8cbd49e](https://github.com/IndexCoop/index-coop-smart-contracts/pull/173/commits/8cbd49e07fd4a9dfae42668b4845610f9ee5445a)